optimum/exporters/executorch/integrations.py

```
@@ -85,13 +85,10 @@ def export(self, input_ids=None, attention_mask=None) -> Dict[str, ExportedProgr
             torch.ones((batch_size, seq_length), dtype=torch.long) if attention_mask is None else attention_mask
         )
 
-        # Define dynamic dimensions using Dim class
-        seq_dim = torch.export.Dim("sequence_length", min=1, max=max_seq_length)  # Allow sequences up to max_length
-
         # Define dynamic shapes with Dim objects
         dynamic_shapes = {
-            "input_ids": {1: seq_dim},
-            "attention_mask": {1: seq_dim},
+            "input_ids": {1: torch.export.Dim.AUTO},
+            "attention_mask": {1: torch.export.Dim.AUTO},
```

@tugsbayasgalan How does the `Dim.AUTO` know which dim need to be dynamic and how the dynamic range is set correctly? For example, if we export with static cache and max_cache_len is fixed, so the max_seq_len can't be larger than the max_cache_len. Same for the encoder-decoder, the seq_len is bounded by the max_hidden_embedding in the models' config, it can't be arbitrary large.

---

After this PR lands, https://github.com/pytorch/pytorch/pull/149590#issuecomment-2738879408 we can specify min/max range for Dim.AUTO. Right now this bypasses error where the seq_length gets specialized to 511.

---

> Right now this bypasses error where the seq_length gets specialized to 511.

@tugsbayasgalan From which model? IIRC, some models seq_length is bounded by `max_position_embeddings`, which is 512. If the seq_length >=512, it will error out. If you silent the error and export with seq_length > 512, I don't think the model can function correctly.

---

```
torch.fx.experimental.symbolic_shapes.ConstraintViolationError: Constraints violated (sequence_length)! For more information, run with TORCH_LOGS="+dynamic".
  - Not all values of sequence_length = L['args'][0][0].size()[1] in the specified range sequence_length <= 511 are valid because sequence_length was inferred to be a constant (511).
  - Not all values of sequence_length = L['args'][1]['attention_mask'].size()[1] in the specified range sequence_length <= 511 are valid because sequence_length was inferred to be a constant (511).
Suggested fixes:
  sequence_length = 511
  ```
  
  This is from deberta_v2 model. Dim.AUTO won't silence the error. It will start with dynamic shapes and keep refining the range until it gets acceptable. So if I specify the max to be 512, it should still work.

---

> It will start with dynamic shapes and keep refining the range until it gets acceptable

This is interesting!

---

(commenting for context) I added Dim.AUTO for export, and it's doing "automatic dynamic shapes", where it collects constraints it sees in the model during tracing (e.g. min/max bounds, dimensions being static, one dimension being equal to another). Export then produces the most dynamic graph possible under those constraints, so you'll always comply with issues like the one you mentioned.

I think the only downside of not being able to specify a max bound, is that the upper bound will be infinite if the model doesn't restrict it, and I guess that doesn't work great with ET memory planning. But that PR is in progress.